Code/Source/URDF/FbxGenerator.cpp

```diff
@@ -0,0 +1,425 @@
+            properties.AddChild(Node("P", {"CoordAxisSign", "int", "Integer", "", 1}));
+            properties.AddChild(Node("P", {"OriginalUpAxis", "int", "Integer", "", 1}));
+            properties.AddChild(Node("P", {"OriginalUpAxisSign", "int", "Integer", "", 1}));
+            properties.AddChild(Node("P", {"UnitScaleFactor", "double", "Number", "", 1}));
+            properties.AddChild(Node("P", {"OriginalUnitScaleFactor", "double", "Number", "", 1}));
+            properties.AddChild(Node("P", {"AmbientColor", "ColorRGB", "Color", "", 0, 0, 0}));
+            properties.AddChild(Node("P", {"DefaultCamera", "KString", "", "", "Producer Perspective"}));
+            properties.AddChild(Node("P", {"TimeMode", "enum", "", "", 11}));
+            properties.AddChild(Node("P", {"TimeProtocol", "enum", "", "", 2}));
+            properties.AddChild(Node("P", {"SnapOnFrameMode", "enum", "", "", 0}));
+            properties.AddChild(Node("P", {"TimeSpanStart", "KTime", "Time", "", 1924423250}));
+            properties.AddChild(Node("P", {"TimeSpanStop", "KTime", "Time", "", 1924423250}));
+            properties.AddChild(Node("P", {"CustomFrameRate", "double", "Number", "", -1}));
+            properties.AddChild(Node("P", {"TimeMarker", "Compound", "", ""}));
+            properties.AddChild(Node("P", {"CurrentTimeMarker", "int", "Integer", "", -1}));
+
+            globalSettings.AddChild(std::move(properties));
+
+            return globalSettings;
+        }
+
+        Node FbxGenerator::GetDocuments() const
+        {
+            Node documents("Documents");
+            documents.AddChild("Count", 1);
+
+            Node document("Document", {"", "Scene"});
+
+            Node properties("Properties70");
+            properties.AddChild(Node("P", {"SourceObject", "object", "", ""}));
+            properties.AddChild(Node("P", {"ActiveAnimStackName", "KString", "", "Take 001"}));
+
+            document.AddChild(std::move(properties));
+            document.AddChild("RootNode", 0);
+            return documents;
+        }
+
+        Node FbxGenerator::GetDefinitions() const
+        {
+            Node definitions("Definitions");
+            definitions.AddChild("Version", 100);
+            definitions.AddChild("Count", 3);
+
+            Node globalSettings("ObjectType", {"GlobalSettings"});
+            globalSettings.AddChild("Count", 1);
+            definitions.AddChild(std::move(globalSettings));
+
+            Node model("ObjectType", {"Model"});
+            model.AddChild("Count", 1);
+            definitions.AddChild(std::move(model));
+
+            Node geometry("ObjectType", {"Geometry"});
+            geometry.AddChild("Count", 1);
+            definitions.AddChild(std::move(geometry));
+
+            Node material("ObjectType", {"Material"});
+            material.AddChild("Count", 1);
+            definitions.AddChild(std::move(material));
```

Comment:
Ok, I will add a method for example `AddNodeWithOneChild` or something like that. I haven't done it before because I wasn't sure if it's not more readable this way.